x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractor.java

```diff
@@ -244,6 +245,26 @@ public DataSummary collectDataSummary() {
         return new DataSummary(searchResponse.getHits().getTotalHits().value, context.extractedFields.getAllFields().size());
     }
 
+    public void collectDataSummaryAsync(ActionListener<DataSummary> dataSummaryActionListener) {
+        SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client, SearchAction.INSTANCE)
```

Comment:
It'd be nice to factor out `SearchRequestBuilder searchDataSummary()` so that if we need to change the search we only need to do it in a single place.